### PR TITLE
PropertyTypeInfo introduced

### PR DESCRIPTION
**Pull request by IlyaKhD**
All property type definition methods come now together: PropertyTypeInfo class is responsible for providing the Generator with info about property CLR type and whether it is a DOM template or a raw-string.

### CODE REVIEW

File: DevExtreme.AspNet.TagHelpers.Generator/PropTypeRegistry.cs [hunk 1]
```diff
@@ -6,99 +6,79 @@
...
-            CLR_STRING = "string",
-            CLR_DATE = "DateTime",
-            CLR_OBJECT = "object",
-            CLR_ARRAY_STRING = "IEnumerable<string>",
-            CLR_ARRAY_OBJECT = "IEnumerable<object>",
-
-            SPECIAL_RAW_STRING = "sp_raw_string",
-            SPECIAL_DOM_TEMPLATE = "sp_dom_template";
 
         public static readonly IDictionary<string, string> OverrideTable = new Dictionary<string, string> {
```
Comment (AlekseyMartynov) [inline DevExtreme.AspNet.TagHelpers.Generator/PropTypeRegistry.cs]: This data can be moved to the new PropertyTypeInfo class and kept private. Hey, in OOP data can stay together with code!

File: DevExtreme.AspNet.TagHelpers.Generator/PropertyTypeInfo.cs [hunk 2]
```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace DevExtreme.AspNet.TagHelpers.Generator {
+
+    public class PropertyTypeInfo {
```
Comment (AlekseyMartynov) [inline DevExtreme.AspNet.TagHelpers.Generator/PropertyTypeInfo.cs]: We should decide whether we use "Property" or "Prop".

File: DevExtreme.AspNet.TagHelpers.Generator/PropTypeRegistry.cs (hunk 1 above)
Comment (IlyaKhD) [inline DevExtreme.AspNet.TagHelpers.Generator/PropTypeRegistry.cs]: I had the same idea, but then thought tht all other manual overrids are in registries, so it looks more consitent to keep this in `PropTypeRegistry` for now.

Comment (AlekseyMartynov) [inline DevExtreme.AspNet.TagHelpers.Generator/PropTypeRegistry.cs]: > keep this in PropTypeRegistry for now

why postpone?

Comment (IlyaKhD) [inline DevExtreme.AspNet.TagHelpers.Generator/PropTypeRegistry.cs]: Hm, I suppose these registries are also be restructured, but you're right - each step should be completed, so will move this to the `PropertyTypeInfo`.

File: DevExtreme.AspNet.TagHelpers.Generator/PropertyTypeInfo.cs (hunk 2 above)
Comment (IlyaKhD) [inline DevExtreme.AspNet.TagHelpers.Generator/PropertyTypeInfo.cs]: Let's use "Prop".

File: DevExtreme.AspNet.TagHelpers.Generator/TagPropertyInfo.cs [hunk 3]
```diff
@@ -79,7 +32,6 @@ class TagPropertyInfo {
                 return "data" + Utils.ToKebabCase(name.Substring(4));
             return null;
         }
-
```
Comment (AlekseyMartynov) [inline DevExtreme.AspNet.TagHelpers.Generator/TagPropertyInfo.cs]: Space-hunting! :smile:

**Comment by AlekseyMartynov** [comment]
Suggested rename:
`GetJsType` => `GetRawType`
`jsTypeString` => `rawTypeString`

File: DevExtreme.AspNet.TagHelpers.Generator/TagPropertyInfo.cs (hunk 3 above)
Comment (IlyaKhD) [inline DevExtreme.AspNet.TagHelpers.Generator/TagPropertyInfo.cs]: Yep

**Comment by IlyaKhD** [comment]
Ok, let's rename these.

**Comment by AlekseyMartynov** [comment]
:shipit:

**Pull request merged**